py-polars/tests/test_lazy.py

```diff
@@ -589,6 +589,7 @@ def test_fill_nan() -> None:
         .collect()["a"]
         .series_equal(pl.Series("a", [1.0, 2.0, 3.0]))
     )
+    assert tuple(df.lazy().fill_nan(None).collect()["a"]) == (1.0, None, 3.0)
```

Comment:
If you don't convert to `tuple` what is the mismatch? Is it a datatype problem?

Reply:
The print methods are identical, but comparision yields with not identical. You can also run the example I posted in my initial PR description to see the problem.

Reply:
Ah, I see. This is a problem with `Series.series_equal`. 

```python
In [25]: s = pl.Series("a", [1.0, None, 3.0])

In [26]: s.series_equal(s)
Out[26]: False
```

I will open a separate issue.

Reply:
@lorenzwalthert `Series.series_equal` has a `null_equal` parameter that defaults to `False`. Can you change the `assert` statement to 

```python
assert (
        df.lazy()
        .fill_nan(None)
        .collect()["a"]
        .series_equal(pl.Series("a", [1.0, None, 3.0]), null_equal=True)
    )
```

Reply:
Ups of course. Did not know that, thanks.

Reply:
So if `null_equal` defaults to `True` for data frames in `frame_equal()`, shouldn't it also have the same default value for `polars.series`?

Reply:
Yes, it should. That's more consistent.